### DISCUSSION TITLE
[DISCUSSION]: PPC toolchain on PS2SDK. Do we want it?

### DISCUSSION THREAD

**Discussion by israpps** [ps2dev/ps2sdk#576] [Polls]
since @qnox32 has been doing some amazing job on PPC stuff, I wondered if we will eventually put the PPC toolchain and compiler here on PS2SDK. what do you think guys?

[ref](https://github.com/israpps/PPC-IOP/tree/main/buildscripts)

**Comment by rickgaiser** [comment]
I think we should add the toolchain (binutils+gcc) as soon as it can build something usefull. I'm not up to date with the latest developments, but perhaps there's already usefull code to be compiled?

**Comment by israpps** [comment]
> I think we should add the toolchain (binutils+gcc) as soon as it can build something usefull. I'm not up to date with the latest developments, but perhaps there's already usefull code to be compiled?

Well, nox has made a program (also hosted on the linked repo) called PPC monitor, basically it's an UART operated program running on the PPC to manipulate the IOP RAM and PPC compatibility modes and more.

Nox even mentioned that a basic debugger is on his plans.

Another ancient project, the hdpro clone of wisi, used a similar software (actually, PPC monitor is partially based on that one)

**Comment by rickgaiser** [comment]
Then it's time to add the PPC toolchain ;). I think whats needed is:
- new repository `ps2toolchain-iop-ppc`
- `ps2toolchain` modified
- `binutils-gdb` can use latest release (no special changes needed?)
- `gcc` can use latest release (no special changes needed?)

Anything I miss?

**Comment by israpps** [comment]
> Anything I miss?

Not sure TBH. all I know is that [these scripts](https://github.com/israpps/PPC-Monitor/tree/main/buildscripts/toolchain) alredy build and install toolchain and gcc for powerpc into `$PS2DEV/powerpc/` with `powerpc-ps2-eabi-` prefix

**Comment by fjtrujy** [comment]
I have migrated this to be a discussion, please proceed.
Cheers
